File: tests/fixtures/cpp/src/timing.cpp
#include "timing.h"
#include <cstdio>
#include <unistd.h>

namespace timing {

void fast() {
    volatile int x = 0;
    for (int i = 0; i < 100; i++) x += i;
}

void medium() {
    volatile double result = 0;
    for (int i = 0; i < 100000; i++) {
        result += i * 0.001;
    }
}

void slow() {
    usleep(50000); // 50ms
}

void very_slow() {
    usleep(500000); // 500ms
}

} // namespace timing

File: tests/fixtures/python/modules/timing.py
"""Timing functions with varied durations."""
import time

def fast():
    time.sleep(0.001)  # 1ms

def medium():
    time.sleep(0.05)   # 50ms

def slow():
    time.sleep(0.1)    # 100ms

def very_slow():
    time.sleep(0.3)    # 300ms
